portfolio/src/main/java/com/google/sps/data/MergeBooks.java

```diff
@@ -0,0 +1,158 @@
+package com.google.sps.data;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.sps.data.Book;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+public class MergeBooks {
+  /**
+   * This method merges two books that have the same title
+   * and ISBN and returns a new book combined from both of them
+   * @param original: the original book
+   * @param addition: the book with additional info you
+   * want to add
+   * The order of the params does not matter
+   */
+  public static Book merge(Book original, Book addition) {
+    validate(original);
+    validate(addition);
+    String isbn = getNonNullString(original.isbn(),
+        addition.isbn()); // One of the ISBNs will be null so you only need to have one valid isbn
+    validateISBN(isbn);
+    Preconditions.checkArgument(
+        original.title().equals(addition.title()), "The books must be of the same title");
+
+    Book.Builder combined_builder = Book.builder();
+    combined_builder.title(original.title());
+    // Combine lists and Sets
+    combined_builder.categories(combineLists(original.categories(), addition.categories()));
+    combined_builder.genre(combineSets(original.genre(), addition.genre()));
+    combined_builder.authors(combineLists(original.authors(), addition.authors()));
+    ArrayList<String> reviews = combineLists(original.reviews(), addition.reviews());
```

Comment:
You shouldn't need custom methods to combine lists/sets. Please update others to use: `ImmutableList.builder().addAll(original.reviews()).addAll(addition.reviews()`

This way, you can also avoid references to ArrayList and replace them with ImmutableList.